modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -30,10 +30,15 @@ var (
 		Name:      "metrics_generator_processor_service_graphs_dropped_spans",
 		Help:      "Number of dropped spans.",
 	}, []string{"tenant"})
-	metricExpiredSpans = promauto.NewCounterVec(prometheus.CounterOpts{
+	metricTotalEdges = promauto.NewCounterVec(prometheus.CounterOpts{
 		Namespace: "tempo",
-		Name:      "metrics_generator_processor_service_graphs_expired_spans",
-		Help:      "Number of spans that expired before finding their pair",
+		Name:      "metrics_generator_processor_service_graphs_edges",
+		Help:      "Total number of edges detected",
```

Comment:
This metric is a bit confusing to me. Are these total number of edges recorded or only unique edges?

Reply:
My intention is keep track of the total amount of unique edges. That should give more context around the amount of expired edges. With this metric we can calculate %age of exipred edges as `total / expired`.

That said, I see the code is not correct and it's currently counting the amount of spans that contribute to an edge... Let me fix that...